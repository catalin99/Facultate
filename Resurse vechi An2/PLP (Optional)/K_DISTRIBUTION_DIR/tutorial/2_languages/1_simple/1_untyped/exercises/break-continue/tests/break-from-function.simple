function f() {
  print("Trying to break from within a function ...\n");
  break;  // normally, this should not be allowed
}

function main() {
  while(true) {
    f();
    print("Should never reach this point.\n");
  }
  print("... and, wrongly, succeeding.\n");
  /* One way to solve this problem dynamically is to empty the loop stack
     when calling a function, but that is expensive, non-modular, and not
     really justified.  Indeed, the clean solution here is to reject such
     programs statically, but we have no static checker for untyped SIMPLE
     (it can be done as part of the static semantics of the typed SIMPLE,
     though).  For now, we can assume that programs do not have such uncaught
     break/continue statements in functions.  */
}
