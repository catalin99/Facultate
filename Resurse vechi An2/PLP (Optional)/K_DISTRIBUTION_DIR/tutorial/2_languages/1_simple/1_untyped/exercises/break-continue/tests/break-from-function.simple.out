Trying to break from within a function ...
... and, wrongly, succeeding.
